src/main/java/net/dv8tion/jda/internal/entities/ReceivedMessage.java

```diff
@@ -157,6 +159,93 @@ public boolean isPinned()
         return getTextChannel().clearReactionsById(getId());
     }
 
+    @Nonnull
+    @Override
+    public RestAction<Void> removeReaction(@Nonnull Emote emote)
+    {
+        return channel.removeReactionById(getId(), emote);
+    }
+
+    @Nonnull
+    @Override
+    public RestAction<Void> removeReaction(@Nonnull Emote emote, @Nonnull User user)
+    {
+        return getTextChannel().removeReactionById(getIdLong(), emote, user);
+    }
+
+    @Nonnull
+    @Override
+    public RestAction<Void> removeReaction(@Nonnull String unicode)
+    {
+        return channel.removeReactionById(getId(), unicode);
+    }
+
+    @Nonnull
+    @Override
+    public RestAction<Void> removeReaction(@Nonnull String unicode, @Nonnull User user)
+    {
+        return getTextChannel().removeReactionById(getId(), unicode, user);
+    }
+
+
+    @Nonnull
+    @Override
+    public ReactionPaginationAction retrieveReactionUsers(@Nonnull Emote emote)
+    {
+        Checks.notNull(emote, "Emote");
+
+        MessageReaction reaction = this.reactions.stream()
+            .filter(r -> r.getReactionEmote().isEmote() && r.getReactionEmote().getEmote().equals(emote))
+            .findFirst().orElse(null);
+
+        Checks.check(reaction != null, "Emoji is present not on this message.");
+        return new ReactionPaginationActionImpl(reaction);
+    }
+
+    @Nonnull
+    @Override
+    public ReactionPaginationAction retrieveReactionUsers(@Nonnull String unicode)
+    {
+        Checks.noWhitespace(unicode, "Emoji");
+
+        MessageReaction reaction = this.reactions.stream()
+            .filter(r -> r.getReactionEmote().isEmoji() && r.getReactionEmote().getEmoji().equals(unicode))
+            .findFirst().orElse(null);
+
+        Checks.check(reaction != null, "Emoji is not present on this message.");
+        return new ReactionPaginationActionImpl(reaction);
+    }
+
+    @Override
+    public MessageReaction.ReactionEmote getReactionByName(@Nonnull String name)
+    {
+        Checks.notNull(name, "Reaction name");
+
+        return this.reactions.stream()
+            .map(MessageReaction::getReactionEmote)
+            .filter(r -> r.getName().equals(name))
+            .findFirst().orElse(null);
+    }
+
+    @Override
+    public MessageReaction.ReactionEmote getReactionById(@Nonnull String id)
+    {
+        Checks.isSnowflake(id, "Reaction ID");
+
+        return getReactionById(MiscUtil.parseSnowflake(id));
```

Comment:
You double check the snowflake here, parsing will throw if its not a snowflake so the extra check is unnecessary.